cmd/terramate-lsp/main.go

```diff
@@ -44,17 +60,24 @@ func main() {
 	// TODO(i4k): implement other modes.
 	if *modeFlag != "stdio" {
 		fmt.Println("terramate-lsp only supports stdio mode")
-		os.Exit(0)
+		os.Exit(1)
 	}
 
+	configureLogging(*logLevelFlag, *logFmtFlag, os.Stderr)
 	runServer(&readWriter{os.Stdin, os.Stdout})
 }
 
 func runServer(conn io.ReadWriteCloser) {
+	logger := log.With().
+		Str("action", "main.runServer()").
+		Logger()
+
+	logger.Trace().Msg("Creating context for UNIX signals.")
+
 	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill, syscall.SIGTERM)
 	defer stop()
 
-	log.Printf("Starting Terramate Language Server in %s mode ...", *modeFlag)
+	logger.Info().Msgf("Starting Terramate Language Server in %s mode ...", *modeFlag)
```

Comment:
```suggestion
	logger.Info().
		Str("mode", *modeFlag).
		Msg("Starting Terramate Language Server")
```

Reply:
I'm gonna apply this but not sure if this is a good candidate to log metadata.

Reply:
you are logging the metadata anyway, the question is to defer the formatting of the metadata informatiion to the log library or do the formatting yourself, since we are already on structured logging land I don't see much of a point on doing the formatting manually but if you think it is better it is all good too, not a huge deal.